### PR TITLE
feat: Add Docker support for disability-max-ratings-api

### PR DESCRIPTION
**Pull request by gabezurita**
# Add Docker Support for Disability Max Ratings API

**Closes:** [#3833](https://github.com/department-of-veterans-affairs/abd-vro/issues/3833)

This PR adds Docker support for the disability-max-ratings-api service, providing optimized configurations for both production and development environments.

## Changes

- **Dockerfile:**  
  - Multi-stage build for reduced production image size  
  - Runs as non-root user for improved security  
  - Minimal dependencies for production

- **docker-compose.yml:**  
  - Separate services for production (`api`) and development (`api-dev`)  
  - Health monitoring via a dedicated `/health` endpoint

- **.dockerignore:**  
  - Improves build performance by excluding unnecessary files

- **pyproject.toml:**  
  - Updated Poetry configuration with comprehensive metadata

- **README.md:**  
  - Detailed instructions for building, running, and testing the service in both environments

## Production Setup

- Uses Python 3.12.3 slim image  
- Employs multi-stage builds for smaller images  
- Health checks enabled  
- Security features: non-root user, `no-new-privileges`

## Development Setup

- Includes all development tools and dependencies  
- Local directory mounting for live code changes  
- Supports running tests and dev tasks directly in the container

## Testing

1. **Build & Start Services**
   ```bash
   docker compose down
   docker compose build --no-cache
   docker compose up -d
   docker compose ps
   ```
   Expected: `disability-max-ratings-api` running on port 8130

2. **Check Endpoints**
   ```bash
   # API docs
   curl http://localhost:8130/docs

   # Health endpoint
   curl http://localhost:8130/health

   # Main endpoint

    curl -X POST 'http://localhost:8130/disability-max-ratings' -H 'accept: application/json' -H 'Content-Type: application/json' -d '{"diagnostic_codes": [6260]}'

   ```
   Expected main endpoint response:
   ```json
   {
       "ratings": [
           { "diagnostic_code": 6260, "max_rating": 10 }
       ]
   }
   ```

3. **Development Environment**
   ```bash
    # Run tests inside container using Poetry
    docker compose run --rm api poetry run pytest
    
    # Check user
    docker compose run --rm api id
   ```
   Expected: All tests pass (>80% coverage), user should be non-root (uid=1000)

4. **Production Environment**
   ```bash
   docker compose exec api id
   docker compose ps
   ```
   Expected: Non-root user in production container, container status "healthy"

5. **Local Development Checks**
   ```bash
   poetry run pre-commit run --all-files
   ```

6. **Cleanup**
   ```bash
   docker compose down
   ```

## Notes

- Pre-commit checks run locally (not in container)
- The `api-dev` service is optional and only needed for development tasks
- Changes in local files are instantly reflected inside the dev container

### CODE REVIEW

**Review by brostk — DISMISSED** [review]

File: .python-version
[the remark this answers is not in the document]
Comment (gabezurita) [reply .python-version:1]: This repo doesn't strictly need it, given that Docker and Poetry use their Python environments, but it does not harm. It only adds real value for developers who want to check the project's Python version quickly. Should it be kept or removed?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+WORKDIR /app
+
+# Install system dependencies required for building
+RUN apt-get update && \
+    apt-get install -y curl && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install Poetry
+RUN pip install --no-cache-dir poetry==1.7.1
```
[the remark this answers is not in the document]
Comment (gabezurita) [reply Dockerfile]: Fixed as a part of this commit!
https://github.com/department-of-veterans-affairs/disability-max-ratings-api/pull/23/commits/fc02911d2bec3ba6d3bdbf6be498389962033d3d

**Comment by gabezurita** [comment]
@dfitchett and @brostk, this is ready for review! Please run the testing steps on your machine and let me know if the Docker setup makes sense.

File: .python-version
[the remark this answers is not in the document]
Comment (dfitchett) [reply .python-version:1]: I'd say lets remove it and let `pyproject.toml` be the source of truth.

**Review by dfitchett — APPROVED** [review]

**Review by dfitchett — APPROVED** [review]

**Pull request merged**